tasks/retrieval_deal.go

```diff
@@ -85,15 +89,32 @@ func (de *retrievalDealExecutor) executeAndMonitorDeal(ctx context.Context, upda
 	}
 
 	dealStage = AddLog(dealStage, "deal sent to miner")
-	err = updateStage("ProposeDeal", dealStage)
+	err = updateStage(stage, dealStage)
 	if err != nil {
 		return err
 	}
 
 	lastStatus := retrievalmarket.DealStatusNew
-	var lastBytesReceived uint64 = 0
+	var lastBytesReceived uint64
+	var prevStage string
+	var stageTimeout <-chan time.Time
+
 	for {
+		if stage != prevStage {
+			// If there is a timeout for the current deal stage, then set the timer
+			if timeout, ok := stageTimeouts[strings.ToLower(stage)]; ok {
+				stageTimeout = time.After(timeout)
+			} else {
+				stageTimeout = nil
```

Comment:
can we add a flag for a default stage timeout so we don't get stuck in an unknown stage forever?

Reply:
This is done by setting `--stage-timeout default:30m`.  Noted in flag usage.